python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -791,3 +812,51 @@ def info_string(self):
         lm_summary = self.load_metrics.summary()
         autoscaler_summary = self.summary()
         return "\n" + format_info_string(lm_summary, autoscaler_summary)
+
+
+class NodeTracker:
+    """Map nodes to their corresponding logs.
+
+    We need to be a little careful here. At an given point in time, node_id <->
+    ip can be interchangeably use, but the node_id -> ip relation is not
+    bijective _across time_ since IP addresses can be reused. Therefore, we
+    should treat node_id as the only unique identifier.
+
+    """
+
+    def __init__(self, log_dir, process_runner):
```

Comment:
when the node becomes active, shouldn't we remove the entry ?
when autoscaler crashes and restarts, shouldn't this call non_terminated_nodes or something?

Reply:
No need, plus this way we're just reusing the process runner in the recovery case